src/counts.jl

```diff
@@ -286,14 +289,15 @@ end
 # faster results and less memory usage. However we still wish to enable others
 # to write generic algorithms, therefore the methods below still accept the 
 # `alg` argument but it is ignored.
-function addcounts!(cm::Dict{Bool}, x::AbstractArray{Bool}; alg = :ignored)
+function _addcounts!(::Type{Bool}, cm::Dict{Bool}, x::AbstractArray{Bool}; alg = :ignored)
     sumx = sum(x)
     cm[true] = get(cm, true, 0) + sumx
     cm[false] = get(cm, false, 0) + length(x) - sumx
     cm
 end
+_addcounts!(::Type{Bool}, cm::Dict{Bool}, x; alg = :ignored) = _addcounts!(Bool, cm, collect(x); alg = alg)
```

Comment:
Again this is going to allocate a copy that could be avoided. I can see two solutions:
- Extend the signature of the method below to `T <: Union{Bool, UInt8, UInt16, Int8, Int16}`. It seems to work fine, it's just slightly suboptimal as `counts` will be of length 256 -- but that's not a big deal.
- Copy the approach used by `_addcounts!(::Type{Bool}, cm::Dict{Bool}, x::AbstractArray{Bool})` above, but without using `length`. A simple `for` loop should be enough.

Reply:
if we're doing loops anyways, can't we just let it fall-back to `addcounts_dict!`?

Reply:
A dict lookup is much slower than an addition. TBH I'm not sure how common it is to call `countmap` on a vector of `Bool`, but since we provide this optimization for arrays...